lib/utils/utf8.c

```diff
@@ -79,3 +84,41 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+bool utf8_is_valid(const unsigned char *str, size_t len)
+{
+	size_t i = 0, nbyte = 0;
+
+	/* It will also return false */
+	if (str == NULL) {
+		return false;
+	}
+
+	while (i < len) {
+		if (str[i] == '\0') {
+			break;
+		}
+		if (str[i] <= ASCII_CHAR && str[i] > '\0') {
```

Comment:
Style: the `> '\0'` condition is disallowed by your test above and is dead code, eliminate.

Reply:
I've made change :)